tests/test_md.py

```diff
@@ -789,7 +789,7 @@ def test_setup_context_with_references():
     seed = 2022
     pressure = constants.DEFAULT_PRESSURE
 
-    unbound_potentials, sys_params, masses, coords, box = get_solvent_phase_system(mol, ff, 0.0, minimize_energy=False)
+    unbound_potentials, sys_params, masses, coords, box = get_solvent_phase_system(mol, ff, 1.0, minimize_energy=False)
```

Comment:
nit: surprising diff -- is this related to the main changes?

Reply:
Yes, it's because having the ligand at lambda 0.0 results in a clash which breaks the test while previously the barostat wouldn't see a clash and accept bad moves

Reply:
Woah -- thanks for clarifying!

Reply:
wtf really?

Reply:
Yes, that is because the minimization is disabled. For the purposes of the test, minimization just slows it down and doesn't impact the correctness. We could also set `minimize_energy=True` and use `lambda=0.0`, at the price of speed.

We could also disable the check that the box volume changed, but that seems useful to test when we are checking if shared pointers are doing what they should.

Reply:
Thanks for the clarification - so the previous barostat test was just broken?

Reply:
Exactly, the test was previously broken.